google/cloud/aiplatform/jobs.py

```diff
@@ -2491,10 +2496,13 @@ def update(
                     deployed_model_ids=deployed_model_ids,
                 )
             )
-        self.api_client.update_model_deployment_monitoring_job(
+        operation = self.api_client.update_model_deployment_monitoring_job(
             model_deployment_monitoring_job=current_job,
             update_mask=field_mask_pb2.FieldMask(paths=update_mask),
         )
+        # TODO: b/254285776 add optional_sync support to model monitoring job
+        self._block_until_complete()
```

Comment:
This shouldn't be necessary. We should only rely on the operation completing through `operation.result()`.

Reply:
Actually, I looked into this again and we should not block the update call at all. I had an offline chat with Jing a while back and she said that depending on what is requested in the update operation, it could take anywhere between minutes and hours for it to complete (by completing I'm referring to the job state going from UPDATING back to RUNNING). It does not make sense for the SDK to block until the operation is complete, especially when the update request is for something time consuming like updating the training dataset used for skew detection. 

The proto representation of the job in the update response will already contain the latest configuration, so it's sufficient to just re-assign that back to `self._gca_resource`. As for when the MDM job will resume running with the updated configs, that's a service-side behavior. The GAPIC code only schedules for the update operation to be executed and does not block on complete, so I don't think the SDK should deviate from that behavior. It also isn't necessary from a UX perspective because developers most likely wouldn't need to do anything else on the monitoring job immediately after updating it. As of right now, if users were to do something immediately after `job.update`, for instance calling `job.pause`, GAPIC will return a gRPC error telling the users that the job cannot be paused when it's in `UPDATING` state. So I think the function behavior is correct as-is.